### ISSUE TITLE
Add patern to placeholder

### ISSUE THREAD

**Issue by orblazer** [filoghost/HolographicDisplays#211]
Hello,
I make that issue for propose implement pattern for custom placeholder (like `world` or `bungeecord` placeholder).
This is useful for pass variable to placeholder.

For me i see that usage :
1. With matcher
    ```java
    HologramsAPI.registerPlaceholder(plugin, pattern, refresfreshRate, (matcher) -> {});
    ```
2. With directly matches
    ```java
    HologramsAPI.registerPlaceholder(plugin, pattern, refresfreshRate, (matches) -> {});
    ```

The usage that new format :
1. ```java
    HologramsAPI.registerPlaceholder(this, Pattern.compile("\{serverMap\:(\w+)\}"), 1, (matcher) -> {
        return getServerMap(matcher.group(1));
    });
    ```
2. ```java
    HologramsAPI.registerPlaceholder(this, Pattern.compile("\{serverMap\:(\w+)\}"), 1, (matches) -> {
        return getServerMap(matches[1]);
    });
    ```

Thanks.

**Comment by filoghost** [comment]
This is a good idea, and will be most likely implemented in Holographic Displays 3.0. I plan to make a lot of refactoring, that will allow me to easily improve placeholders.

**Comment by filoghost** [comment]
Implemented in the new beta API, it needs to be improved and documented.
